crates/transaction-emitter/src/lib.rs

```diff
@@ -755,14 +780,21 @@ async fn wait_for_accounts_sequence(
     client: &RestClient,
     accounts: &mut [LocalAccount],
     txn_expiration_time_secs: u64,
+    check_for_stats_at_end: bool,
 ) -> Result<(), HashSet<AccountAddress>> {
-    // TXN_EXPIRATION_SECONDS == TXN_MAX_WAIT
-    let wait = Duration::from_secs(min(txn_expiration_time_secs, TXN_EXPIRATION_SECONDS));
+    let wait = match check_for_stats_at_end {
```

Comment:
Maybe it will be cleaner if the caller of this function just pass the `wait time` instead and leave this as a generic function?